ci/infra/testrunner/utils/command.py

```diff
@@ -0,0 +1,44 @@
+class CommandResult:
+    """
+    Embeds the result of running a command for easier testing of different
+    output channels (stdout, stderr) and the return code of the execution.
+
+    Delegates iterators, direct comparisons and unknown attribute retrieval
+    (including function calls) to the `stdout` attribute for better testing
+    ergonomics.
+    """
+    def __init__(self, retcode, stdout, stderr):
+        self.retcode = retcode
+        self.stdout = stdout
+        self.stderr = stderr
+
+    def __eq__(self, other):
+        if isinstance(other, str):
+            return self.stdout == other
+        elif isinstance(other, self.__class__):
+            return self.retcode == other.retcode and self.stdout == other.stdout and self.stderr == other.stderr
+        return False
```

Comment:
Should this also implement integer comparison, or do we assume that such a comparison will probably wrap this in `int()`?  Maybe so; anyone doing `>`, `<`, etc would also use `int()`.

Reply:
For use cases that were doing `int(stdout)` in our test code, we have the cast `__int__` implemented, for comparisons I'd say comparing against an string or against another `CommandResult` object would be enough. Even only with string, the other was implemented for completion, but I doubt of its usefulness honestly.